### ISSUE TITLE
projection design

### ISSUE THREAD

**Issue by changhiskhan** [eto-ai/lance#95]
when we project `annotations.label` for example, the output schema is still a list of struct, so we're still faced as query syntax issues as before.

Would it make sense for the output schema to become just a list of string instead?

### Expected Behavior

For a schema of list of struct, `annotations.label` returns `list<string>` instead of `list<struct<label: string>>`

**Comment by eddyxu** [comment]
Returns `annotations.label` as list has better performance characteristics and less code in training data loader as well. 

For example, a pytorch dataset can directly return `{"labels": [1, 2, 3, 4], "boxes": [[...], [...]]}` instead of `[{"label": 1, "box": [....]}, {"label": 2, "box": [...]}`.

**Comment by changhiskhan** [comment]
Can we do something Xpath-like...but not shitty ? Lol

**Comment by eddyxu** [comment]
Let's try to formalize the requirement here. 

Ignoring the actual SQL engine implementation,  let us just discuss what kind of queries we need support:

Assuming we have a dataset with schema:

```
id: int64
image: image<binary>
split: string
annotations: list<struct<
   label: dict<string, int8>,
   box: box2d,
   comment: string,
   object_id: int64
>>
```

Query 1: Label distribution globally

```sql
SELECT COUNT(1), label FROM (
   SELECT UNNEST(annotations.label) FROM t
) GROUP BY 2
```

Query 2: Calculate of the label distribution between splits

TBD

Query 3: Find images that contain the specific labels.

TBD

Query 4: Find the images and bounding box of specific labels. 

TBD
